R/optionalInput.R

```diff
@@ -137,58 +210,94 @@ optionalSelectInput <- function(inputId, # nolint
     selected <- NULL
   }
 
+  if (length(choices) <= 1 && fixed_on_single) fixed <- TRUE
+
   raw_choices <- extract_raw_choices(choices, attr(choices, "sep"))
   raw_selected <- extract_raw_choices(selected, attr(choices, "sep"))
 
-  ui <- shinyWidgets::pickerInput(
-    inputId = inputId,
-    label = label,
-    choices = raw_choices,
-    selected = raw_selected,
-    multiple = TRUE,
-    width = width,
-    options = options,
-    choicesOpt = picker_options(choices)
+  ui_picker <- tags$div(
+    id = paste0(inputId, "_input"),
+    # visibility feature marked with display: none/block instead of shinyjs::hide/show
+    #  as mechanism to hide/show is handled by javascript code
+    style = if (fixed) "display: none;" else "display: block;",
+    shinyWidgets::pickerInput(
+      inputId = inputId,
+      label = label,
+      choices = raw_choices,
+      selected = raw_selected,
+      multiple = TRUE,
+      width = width,
+      options = options,
+      choicesOpt = picker_options(choices)
+    )
   )
 
   if (!is.null(label_help)) {
-    ui[[3]] <- append(ui[[3]], list(div(class = "label-help", label_help)), after = 1)
+    ui_picker[[3]] <- append(ui_picker[[3]], list(div(class = "label-help", label_help)), after = 1)
   }
 
-  shiny::tagList(
-    include_css_files(pattern = "picker_input"),
-    if (is.null(choices)) {
-      shinyjs::hidden(ui)
-    } else {
-      if (fixed) {
-        div(
-          shinyjs::hidden(ui),
-          tags$label(id = paste0(inputId, "_textonly"), class = "control-label", sub(":[[:space:]]+$", "", label)),
-          if (length(selected) > 0) {
-            tags$code(
-              id = paste0(inputId, "_valueonly"),
-              paste(selected, collapse = ", ")
-            )
-          },
-          label_help
-        )
+  ui_fixed <- tags$div(
+    id = paste0(inputId, "_fixed"),
+    # visibility feature marked with display: none/block instead of shinyjs::hide/show
+    #  as mechanism to hide/show is handled by javascript code
+    style = if (fixed) "display: block;" else "display: none;",
+    tags$label(class = "control-label", sub(":[[:space:]]+$", "", label)),
```

Comment:
Is the `sub` for trimming white space? 
```
label <- "Fixed choices   "
sub(":[[:space:]]+$", "", label)
[1] "Fixed choices   "
```
How about `trimws`?

Reply:
trimws won't work - above removes white space and preceding `:`

Reply:
I don't think it does, actually:
```
label <- "Fixed choices  :    "
sub(":[[:space:]]+$", "", label)
[1] "Fixed choices  "
```

The trailing ones are removed but the preceding ones remain.
How about `"\\s+:\\s+$"`?

Reply:
I don't think it is a good idea to have this gsub at all. Why do we fix a label? It's unintuitive behaviour. Do you mind if I remove this?

Reply:
I certainly don't :slightly_smiling_face: